website/src/pages/StatusPage/StatusPage.tsx

```diff
@@ -32,23 +32,92 @@ export default function StatusPage() {
+    type: '',
+    status: '',
+    source: '',
+  });
+
+  const filterData = (data) => {
+    return data.filter((item) => {
+      const itemType = item.type_of ? item.type_of.toString() : '';
+      const itemSource = item.source ? item.source.toString() : '';
+      const itemStatus = item.status ? item.status.toString() : '';
+
+      const matchesType =
+        filters.type === '' || filters.type === undefined || itemType === filters.type;
+      const matchesSource =
+        filters.source === '' || filters.source === undefined || itemSource === filters.source;
+      const matchesStatus =
+        filters.status === '' || filters.status === undefined || itemStatus === filters.status;
+
+      return matchesType && matchesSource && matchesStatus;
+    });
+  };
+
+  const handleFilterChange = (e) => {
+    const { name, value } = e.target;
+    setFilters((prevFilters) => ({ ...prevFilters, [name]: value }));
+  };
+
+  const filteredData = filterData(dataQueue);
+
   return (
     <>
       <Hero />
 
       <div className="border-accent border mt-5" />
 
+      {/* FILTERS OPTIONS*/}
+      <div>
+        <label>
+          Type:
+          <select name="type" value={filters.type} onChange={handleFilterChange} className="bg-accent rounded-xl p-1">
+            <option value="">All</option>
+            {[...new Set(filteredData.map((item) => item.type_of))].map((type) => (
+              <option key={type} value={type}>
+                {type}
+              </option>
+            ))}
+          </select>
+        </label>
+
+        <label>
+          Source:
+          <select name="source" value={filters.source} onChange={handleFilterChange} className="bg-accent rounded-xl p-1">
+            <option value="">All</option>
+            {[...new Set(filteredData.map((item) => item.source))].map((source) => (
+              <option key={source} value={source}>
+                {source}
+              </option>
+            ))}
+          </select>
+        </label>
+
+        <label>
+          Status:
+          <select name="status" value={filters.status} onChange={handleFilterChange} className="bg-accent rounded-xl p-1">
+            <option value="">All</option>
+            {[...new Set(filteredData.map((item) => item.status))].map((status) => (
+              <option key={status} value={status}>
+                {status}
+              </option>
+            ))}
+          </select>
+        </label>
+      </div>
+
       {/* EXECUTION QUEUE */}
       {!isLoadingQueue && dataQueue && dataQueue.length > 0 && (
-        <ExecutionQueue data={dataQueue} title={"Execution Queue"} />
+        <ExecutionQueue data={filteredData} title={"Execution Queue"} />
       )}
 
       {/* PREVIOUS EXECUTIONS */}
       {!isLoadingPreviousExe &&
         dataPreviousExe &&
         dataPreviousExe.length > 0 && (
           <PreviousExecutions
-            data={dataPreviousExe}
+            data={filteredData}
```

Comment:
You are sending `filteredData` which is filled with `dataQueue` even though it should be filled with `dataPreviousExe`. This leads to having both parts of the page "execution queue" and "previous execution" showing the exact same data.

Reply:
resolved